pkg/controllers/user/globaldns/user_cluster_ingress.go

```diff
@@ -116,59 +117,74 @@ func (ic *UserIngressController) noGlobalDNS() bool {
 	return len(gd) == 0
 }
 
-func gatherIngressEndpoints(ingressEps []v1.LoadBalancerIngress) []string {
-	endpoints := []string{}
-	for _, ep := range ingressEps {
-		if ep.IP != "" {
-			endpoints = append(endpoints, ep.IP)
-		} else if ep.Hostname != "" {
-			endpoints = append(endpoints, ep.Hostname)
+func (ic *UserIngressController) reconcileAllGlobalDNSs() (runtime.Object, error) {
+	globalDNSList, err := ic.globalDNSLister.List("", labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, globalDNSObj := range globalDNSList {
+		//call update on each GlobalDNS obj that refers to this current cluster
+		targetsCluster, err := ic.doesGlobalDNSTargetCurrentCluster(globalDNSObj)
+		if err != nil {
+			return nil, err
+		}
+		if targetsCluster {
+			//enqueue it to the globalDNS controller
+			ic.globalDNSs.Controller().Enqueue(namespace.GlobalNamespace, globalDNSObj.Name)
 		}
 	}
-	return endpoints
+	return nil, nil
+}
+
+func (ic *UserIngressController) doesGlobalDNSTargetCurrentCluster(globalDNS *v3.GlobalDNS) (bool, error) {
+	if globalDNS.Spec.MultiClusterAppName != "" {
+		mcappName, err := getMultiClusterAppName(globalDNS.Spec.MultiClusterAppName)
+		if err != nil {
+			return false, err
+		}
+		mcapp, err := ic.multiclusterappLister.Get(namespace.GlobalNamespace, mcappName)
+		if err != nil {
+			return false, err
+		}
+		// go through target projects and check if its part of the current cluster
+		for _, t := range mcapp.Spec.Targets {
+			split := strings.SplitN(t.ProjectName, ":", 2)
```

Comment:
as lines 152-160 sorta repeat lines 163-171,  we can extract them in some way. Either extract in a separate method, or just do smth like

```
var projectNames []string

if globalDNS.Spec.MultiClusterAppName != "" {
// derive projectNames from MCA and assign to projectNames
} else {
projectNames = globalDNS.Spec.ProjectNames
}
// and then go over projectName logic
 

```

Reply:
done